manager/allocator/cnmallocator/portallocator.go

```diff
@@ -324,9 +324,18 @@ func (pa *portAllocator) isPortsAllocatedOnInit(s *api.Service, onInit bool) boo
 	}
 
 	portStates := allocatedPorts{}
+	hostTargetPorts := map[uint32]struct{}{}
 	for _, portState := range s.Endpoint.Ports {
 		if portState.PublishMode == api.PublishModeIngress {
 			portStates.addState(portState)
+		} else if portState.PublishMode == api.PublishModeHost {
+			// build a map of host mode ports we've seen. if in the spec we get
+			// a host port that's not in the service, then we need to do
+			// allocation. if we get the same target port but something else
+			// has changed, then HostPublishPortsNeedUpdate will cover that
+			// case. don't ask me why this is handled in two different code
```

Comment:
nit: @dperny can you add this as a TODO(dperny) or something so it's easy to search for later?

Reply:
it's not really a TODO, it's just a comment. it's handled correctly in the other case i'm just explaining what it is. i'm adding "don't ask me" because i don't want someone to to a `git blame` in a year and ask my why this is the case.

Reply:
This comment should instead reference #2143.